homeassistant/components/august/lock.py

```diff
@@ -43,27 +42,49 @@ def __init__(self, data, lock):
 
     async def async_lock(self, **kwargs):
         """Lock the device."""
-        update_start_time_utc = dt.utcnow()
-        lock_status = await self.hass.async_add_executor_job(
-            self._data.lock, self._lock.device_id
-        )
-        self._update_lock_status(lock_status, update_start_time_utc)
+        await self._call_lock_operation(self._data.lock)
 
     async def async_unlock(self, **kwargs):
         """Unlock the device."""
-        update_start_time_utc = dt.utcnow()
-        lock_status = await self.hass.async_add_executor_job(
-            self._data.unlock, self._lock.device_id
+        await self._call_lock_operation(self._data.unlock)
+
+    async def _linked_doorsense_entity_id(self):
+        if self._data.lock_has_doorsense(self._lock.device_id):
+            entity_registry = (
+                await self.hass.helpers.entity_registry.async_get_registry()
+            )
+            return entity_registry.async_get_entity_id(
+                "binary_sensor",
+                DOMAIN,
+                find_linked_doorsense_unique_id(self._lock.device_id),
+            )
+        return None
+
+    async def _call_lock_operation(self, lock_operation):
+        activities = await self.hass.async_add_executor_job(
+            lock_operation, self._lock.device_id
         )
-        self._update_lock_status(lock_status, update_start_time_utc)
+        for lock_activity in activities:
+            update_lock_detail_from_activity(self._lock_detail, lock_activity)
+            if isinstance(lock_activity, DoorOperationActivity):
```

Comment:
Also please use guard clauses. `if not isinstance(…): continue`

Reply:
will do